quickwit/quickwit-metastore/src/metastore/file_backed/mod.rs

```diff
@@ -809,6 +809,47 @@ impl MetastoreService for FileBackedMetastore {
         Ok(ServiceStream::new(splits_responses_stream))
     }
 
+    async fn list_index_stats(
+        &self,
+        request: ListIndexStatsRequest,
+    ) -> MetastoreResult<ListIndexStatsResponse> {
+        let index_id_matcher =
+            IndexIdMatcher::try_from_index_id_patterns(&request.index_id_patterns)?;
+        let index_id_incarnation_id_opts: Vec<(IndexId, Option<Ulid>)> = {
+            let inner_rlock_guard = self.state.read().await;
+            inner_rlock_guard
+                .indexes
+                .iter()
+                .filter_map(|(index_id, index_state)| match index_state {
+                    LazyIndexStatus::Active(_) if index_id_matcher.is_match(index_id) => {
+                        Some(index_id)
+                    }
+                    _ => None,
+                })
+                .map(|index_id| (index_id.clone(), None))
+                .collect()
+        };
+
+        let mut index_stats = Vec::new();
+        for (index_id, incarnation_id_opt) in index_id_incarnation_id_opts {
+            match self
+                .read_any(&index_id, incarnation_id_opt, |index| index.get_stats())
```

Comment:
Behind the hood, this is going to open each metastore file and then fetch the split stats sequentially, which is going to be slow if there are many indexes. Try to parallelize this logic using `FuturesUnordered`. There are examples in the file.